pkg/microservice/aslan/core/multicluster/service/clusters.go

```diff
@@ -85,15 +106,77 @@ func GetCluster(id string, logger *zap.SugaredLogger) (*commonmodels.K8SCluster,
 	return s.GetCluster(id, logger)
 }
 
-func CreateCluster(cluster *commonmodels.K8SCluster, logger *zap.SugaredLogger) (*commonmodels.K8SCluster, error) {
-	s, _ := kube.NewService("")
+func convertToAdvancedConfig(nodeSelectorRequirements []*commonmodels.NodeSelectorRequirement) *AdvancedConfig {
+	var nodeLabels []string
+	for _, nodeSelectorRequirement := range nodeSelectorRequirements {
+		for _, labelValue := range nodeSelectorRequirement.Value {
+			nodeLabels = append(nodeLabels, fmt.Sprintf("%s:%s", nodeSelectorRequirement.Key, labelValue))
+		}
+	}
+	advancedConfig := &AdvancedConfig{
+		NodeLabels: nodeLabels,
+	}
+	return advancedConfig
+}
 
-	return s.CreateCluster(cluster, logger)
+func convertToNodeSelectorRequirements(args *AdvancedConfig) []*commonmodels.NodeSelectorRequirement {
+	var nodeSelectorRequirements []*commonmodels.NodeSelectorRequirement
+	nodeLabelM := make(map[string][]string)
+	for _, nodeLabel := range args.NodeLabels {
+		if !strings.Contains(nodeLabel, ":") || len(strings.Split(nodeLabel, ":")) != 2 {
+			continue
+		}
+		key := strings.Split(nodeLabel, ":")[0]
+		value := strings.Split(nodeLabel, ":")[1]
+		nodeLabelM[key] = append(nodeLabelM[key], value)
+	}
+	for key, value := range nodeLabelM {
+		nodeSelectorRequirements = append(nodeSelectorRequirements, &commonmodels.NodeSelectorRequirement{
+			Key:      key,
+			Value:    value,
+			Operator: corev1.NodeSelectorOpIn,
+		})
+	}
+	return nodeSelectorRequirements
 }
 
-func UpdateCluster(id string, cluster *commonmodels.K8SCluster, logger *zap.SugaredLogger) (*commonmodels.K8SCluster, error) {
+func CreateCluster(args *K8SCluster, logger *zap.SugaredLogger) (*commonmodels.K8SCluster, error) {
 	s, _ := kube.NewService("")
+	advancedConfig := new(commonmodels.AdvancedConfig)
+	if args.AdvancedConfig != nil {
+		advancedConfig.Strategy = args.AdvancedConfig.Strategy
+		advancedConfig.NodeLabels = convertToNodeSelectorRequirements(args.AdvancedConfig)
+	}
+	cluster := &commonmodels.K8SCluster{
+		Name:           args.Name,
+		Description:    args.Description,
+		AdvancedConfig: advancedConfig,
+		Status:         args.Status,
+		Production:     args.Production,
+		Provider:       args.Provider,
+		Local:          args.Local,
+		CreatedAt:      args.CreatedAt,
+		CreatedBy:      args.CreatedBy,
+	}
+	if args.ID != "" {
+		cluster.ID, _ = primitive.ObjectIDFromHex(args.ID)
+	}
+	return s.CreateCluster(cluster, logger)
```

Comment:
call a method in mongodb in this way:
`mongodb.NewK8sClusterColl().Create(id, cluster)`
so that you don't need to covert a string to objectID in this file